userspace/libsinsp/plugin.cpp

```diff
@@ -732,7 +732,41 @@ bool sinsp_plugin::resolve_dylib_symbols(void *handle, std::string &errstr)
 
 				if (jvargRequired.asBool() == true)
 				{
-					tf.m_flags = filtercheck_field_flags::EPF_REQUIRES_ARGUMENT;
+					// All the extra casting is because this is the one flags value
+					// that is strongly typed and not just an int.
+					tf.m_flags = (filtercheck_field_flags) ((int) tf.m_flags | (int) filtercheck_field_flags::EPF_REQUIRES_ARGUMENT);
+				}
+			}
+
+			const Json::Value &jvProperties = root[j].get("properties", Json::Value::null);
+			if (!jvProperties.isNull())
+			{
+				if (!jvProperties.isArray())
+				{
+					throw sinsp_exception(string("error in plugin ") + m_name + ": field " + fname + " properties property is not array ");
+				}
+
+				for(Json::Value::ArrayIndex k = 0; k < jvProperties.size(); k++)
+				{
+					const Json::Value &prop = jvProperties[k];
+
+					if (!prop.isString())
+					{
+						throw sinsp_exception(string("error in plugin ") + m_name + ": field " + fname + " properties value is not string ");
+					}
+
+					const std::string &str = prop.asString();
+
+					if(str == "hidden")
+					{
+						tf.m_flags = (filtercheck_field_flags) ((int) tf.m_flags | (int) filtercheck_field_flags::EPF_TABLE_ONLY);
+					}
+					// These are not used by this plugin framework, but are used by other tools like wireshark. So validate the values but
+					// don't do anything with them.
+					else if (str != "conversation" && str != "info")
```

Comment:
Is this check necessary? Since none of this is used in this framework and `libs` consumers, wouldn't it suffice to consider the `hidden` edge case only, and ignore any unused value?

Reply:
It would prevent things like people mistyping hidden (or conversation, or info), but since we'll probably allow the set of properties to evolve faster than we'll change libs, I can remove this check.